src/sources/dnstap/parser.rs

```diff
@@ -0,0 +1,1105 @@
+                );
+            }
+
+            if let Some(query_address) = dnstap_message.query_address {
+                let source_address = if socket_family == 1 {
+                    let address_buffer: [u8; 4] = query_address[0..4].try_into()?;
+                    IpAddr::V4(Ipv4Addr::from(address_buffer))
+                } else {
+                    let address_buffer: [u8; 16] = query_address[0..16].try_into()?;
+                    IpAddr::V6(Ipv6Addr::from(address_buffer))
+                };
+
+                self.insert(
+                    self.event_schema.dnstap_message_schema().query_address(),
+                    source_address.to_string(),
+                );
+            }
+
+            if let Some(query_port) = dnstap_message.query_port {
+                self.insert(
+                    self.event_schema.dnstap_message_schema().query_port(),
+                    query_port as i64,
+                );
+            }
+
+            if let Some(response_address) = dnstap_message.response_address {
+                let response_addr = if socket_family == 1 {
+                    let address_buffer: [u8; 4] = response_address[0..4].try_into()?;
+                    IpAddr::V4(Ipv4Addr::from(address_buffer))
+                } else {
+                    let address_buffer: [u8; 16] = response_address[0..16].try_into()?;
+                    IpAddr::V6(Ipv6Addr::from(address_buffer))
+                };
+
+                self.insert(
+                    self.event_schema.dnstap_message_schema().response_address(),
+                    response_addr.to_string(),
+                );
+            }
+
+            if let Some(response_port) = dnstap_message.response_port {
+                self.insert(
+                    self.event_schema.dnstap_message_schema().response_port(),
+                    response_port as i64,
+                );
+            }
+        }
+
+        if let Some(query_zone) = dnstap_message.query_zone {
+            let mut decoder: BinDecoder = BinDecoder::new(&query_zone);
+            match Name::read(&mut decoder) {
+                Ok(raw_data) => {
+                    self.insert(
+                        self.event_schema.dnstap_message_schema().query_zone(),
+                        raw_data.to_utf8(),
+                    );
+                }
+                Err(error) => return Err(Error::from(error.to_string())),
+            }
+        }
+
+        let dnstap_message_type_id = dnstap_message.r#type;
+        self.insert(
+            &self
+                .event_schema
+                .dnstap_message_schema()
+                .dnstap_message_type_id(),
+            dnstap_message_type_id as i64,
+        );
+
+        self.insert(
+            self.event_schema
+                .dnstap_message_schema()
+                .dnstap_message_type(),
+            format!("{:?}", to_dnstap_message_type(dnstap_message_type_id)?),
```

Comment:
Same here, should we be using `to_string()` instead of the `Debug` format?

Reply:
Will change the code to not use debug format here.

Reply:
Changed.